xclim/sdba/adjustment.py

```diff
@@ -228,16 +253,49 @@ def _train(self, ref, hist):
             description="Scaling factor making the mean of hist match the one of hist.",
         )
 
-    def _adjust(self, sim, degree=0):
+    def _adjust(self, sim, degree=None):
+        # Apply preliminary scaling from obs to hist
         sim = apply_correction(
             sim,
             broadcast(self.ds.scaling, sim, group=self.group, interp=self.interp),
             self.kind,
         )
-        sim_fit = PolyDetrend(degree=degree, kind=self.kind).fit(sim)
-        sim_detrended = sim_fit.detrend(sim)
+
+        # Normalize sim group-wise
+        # This group-wise pre normalization + reapplication further down
+        # is to circumvent #442 (detrending is not groupwise)
+        mu_sim = self.group.apply("mean", sim)
+        sim_norm = apply_correction(
+            sim,
+            broadcast(
+                invert(mu_sim, kind=self.kind),
+                sim,
+                group=self.group,
+                interp=self.interp,
+            ),
+            kind=self.kind,
+        )
+
+        # Find trend on sim (for debugging purpose, the detrending is overridable)
+        if degree is not None:
+            detrending = PolyDetrend(degree=degree, kind=self.kind)
+        else:
+            detrending = self.detrending
```

Comment:
It is very confusing that we can set detrending in the init, and degree in adjust. One API would be to accept any BaseDetrend subclass in adjust, and default to PolyDetrend when an `int` is given.

Reply:
Hum I actually don't agree with having a `degree` parameter  in the `adjust` at all. I think this is only there as a debug workaround for one of the tests.

Reply:
Why ? Let's say you want to compare an adjustment with degree 1 and degree 2, at the moment, you'd have to do something like
```
adj['detrending'] = 2
adj.adjust(sim)
```

Reply:
I was seeing the Adjustment objects as "frozen". You can only adjust with a trained object with one set of parameters. This ensures that, for example, the `adj_params` attribute gives a complete description of the adjustment. For changing the detrending degree I would suggest recreating a whole object and train it on the same ref and hist.

Reply:
What about storing the history with the call from adjust ?
```scen.attrs["history"] = update_history(
            f"Bias-adjusted with method {str(self)}.adjust(sim, **f{kwargs})", sim
)
```
I guess my point is that training is compute intensive, so we should make sure the training object can be reused whenever possible. Otherwise, it doesn't really make sense to split `train` and `adjust`.

Reply:
We could do that. But to stay coherent, all adjust-only parameters should be passable through adjust, no? (namely: extrapolation and interp).
The train-adjust still makes sense as the adjust could be called on many realizations or models,  but you are right that it's less clear.

Reply:
Yes, now that these are done in the adjust stage (thanks to the quantile interpolation), these could be `adjust` parameters.

Reply:
Can you provide a clearer example of what analysis steps would be like under the different scenarios... i.e. frozen vs flexible adjust objects?

Reply:
> I guess my point is that training is compute intensive, so we should make sure the training object can be reused whenever possible. Otherwise, it doesn't really make sense to split `train` and `adjust`.

Aren't training results dependent on the detrend method chosen?  I.e. I don't see how you could change the poly degree without having to retrain your correction factors..